src/main/java/org/broadinstitute/hellbender/dev/pipelines/bqsr/ApplyWholeBQSRTransform.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.hellbender.dev.pipelines.bqsr;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.POutput;
+import htsjdk.samtools.SAMFileHeader;
+import org.broadinstitute.hellbender.dev.tools.walkers.bqsr.BaseRecalibrationArgumentCollection;
+import org.broadinstitute.hellbender.tools.ApplyBQSRArgumentCollection;
+
+/**
+ * Both BQSR phases, together.
+ *
+ * Create the inputs like this:
+ *
+ * PCollection<Read> reads = ...
+ * PCollection<SimpleInterval> knownIntervals = ...
+ * PCollectionTuple inputs =
+ *     PCollectionTuple.of(BaseRecalibratorDataflowUtils.readTag, reads)
+ *                     .and(BaseRecalibratorDataflowUtils.intervalTag, intervals);
+ */
+public class ApplyWholeBQSRTransform extends PTransform<PCollectionTuple, PCollection<Read>> {
```

Comment:
In the process I discovered that saving a report and immediately loading it back is *not* a no-op.
it actually changes them in some necessary way. 

It would be great if someone familiar with that part of the code could help factor the mutation apart from the saving/loading, so I could just call that in the pipeline.